### PR TITLE
Feature/lang logic extract

### PR DESCRIPTION
**Pull request by rozhn0v**
- only valid generated domains are checked
- `xn-idx` is not passed through whole execution path
- extracted method for language checking, `false_true_counter` is not an `int`

### CODE REVIEW

File: ipv4util.py [hunk 1]
```diff
@@ -151,6 +155,14 @@ def get_ip_and_asn(self, hostname: Domain) -> \
             return ip_obj, match_obj.asn
         return ip_obj, None
 
+    @lru_cache(maxsize=128)
```
Comment (yuriarg) [inline ipv4util.py:162]: What does it do?

File: domain.py [hunk 2]
```diff
@@ -71,59 +73,65 @@ def punycode_idx(self) -> List[int]:
             in a dot-separated fashion, where valid punycode can be found.
         """
 
+        def has_xn(string):
+            return 'xn--' in string
+
+        def has_non_ascii(string):
+            return not Domain._is_ascii(string)
+
+        if Domain._is_ascii(self._fqdn):
```
Comment (yuriarg) [inline domain.py:82]: If Python 3.8+ is used, the walrus operator could be used:
`if is_ascii:=Domain._is_ascii(self._fqdn):
        ...
else:
        return not is_ascii
`
This operation to check if a string is ascii encoded iterates letter by letter ?

File: ipv4util.py (hunk 1 above)
Comment (rozhn0v) [reply ipv4util.py:162]: it's a decorator that memoizes results of a method. so last 128 (default value) results will be cached

File: domain.py (hunk 2 above)
Comment (rozhn0v) [reply domain.py:82]: google says it's the fastest way because it utulizes native C codec 😕

**Pull request merged**